src/modules/decentralized-web-node/services/message.service.ts

```diff
@@ -0,0 +1,113 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { FindManyOptions, Repository } from 'typeorm';
+import { Logger } from '../../logger/logger.service';
+import { DataSort } from '../dtos/collections-query-descriptor.dto';
+import { CollectionsWriteMessage } from '../entities/collections-write-message.entity';
+
+@Injectable()
+export class MessageService {
+  constructor(
+    private readonly logger: Logger,
+    @InjectRepository(CollectionsWriteMessage)
+    private readonly messageRepository: Repository<CollectionsWriteMessage>
+  ) {
+    this.logger.setContext(MessageService.name);
+  }
+
+  async find(
+    user: string,
+    options: {
+      dateSort?: DataSort;
+      dataFormat?: string;
+      schema?: string;
+      objectId?: string;
+    }
+  ): Promise<CollectionsWriteMessage[]> {
+    const whereParams: Record<string, string | number> = {
+      target: user,
+    };
+
+    if (options.schema) {
+      whereParams.schema = options.schema;
+    }
+
+    if (options.dataFormat) {
+      whereParams.dataFormat = options.dataFormat;
+    }
+
+    if (options.objectId) {
+      whereParams.objectId = options.objectId;
+    }
+
+    let order: FindManyOptions<CollectionsWriteMessage>['order'];
+    switch (options.dateSort) {
+      case 'createdDescending':
+        order = {
+          dateCreated: 'DESC',
+        };
+        break;
+      case 'publishedAscending':
+        order = {
+          datePublished: 'ASC',
+        };
+        break;
+      case 'publishedDescending':
+        order = {
+          datePublished: 'DESC',
+        };
+        break;
+      default:
+      case 'createdAscending':
+        order = {
+          dateCreated: 'ASC',
+        };
+        break;
+    }
+
+    const items = await this.messageRepository.find({
+      where: whereParams,
+      order,
+    });
+    return items;
+  }
+
+  async save(
+    message: CollectionsWriteMessage,
+    user: string
+  ): Promise<CollectionsWriteMessage | undefined> {
+    const foundedMessage = await this.messageRepository.findOne({
+      where: {
+        objectId: message.objectId,
+        target: user,
+      },
+    });
+    let item: CollectionsWriteMessage = foundedMessage;
+
+    // https://identity.foundation/decentralized-web-node/spec/#processing-instructions
+    switch (true) {
+      case foundedMessage && message.dateCreated > foundedMessage.dateCreated:
+        item = await this.messageRepository.save(message);
+        break;
+
+      case foundedMessage && message.dateCreated < foundedMessage.dateCreated:
+        item = foundedMessage;
+        break;
+
+      case foundedMessage && message.dateCreated === foundedMessage.dateCreated:
+        if (message.cid > foundedMessage.cid) {
+          item = await this.messageRepository.save(message);
+        } else {
+          item = foundedMessage;
+        }
+        break;
+
+      default:
+      case !foundedMessage:
```

Comment:
To me, it's weird to have a case statement after `default`. I think the `case` could be removed without changing the behaviour here, right?
```suggestion
      default:
```